uint32 assetID
string location